docs/_doc/tutorials/wurst_for_vjass_users/page_1.md

```diff
@@ -0,0 +1,234 @@
+
+That cool LUA thing where object data gets accessed but kind of doesn't work these days?
+
+Yeah, wurst has that.
+It's just part of wurst.
+Honestly I'm so spoiled I don't even think about these things anymore.
+
+## On performance
+
+This subject is scary complicated, but it's something people often bring up when discussing vJass and indeed wurst.
+
+Let's get some basics down:
+
+* Coders interested in performance tend to appreciate that in vJass, the shape of the emitted jass tends to be predictable and close to to the low level function designed.
+  This lets them reason about performance.
+* Higher level languages like C++ use an *optimizer* to improve performance by doing things like automatically unraveling loops, inlining functions, and throwing away unused data or code paths.
+* In the past, higher level languages that still provide system-level interaction tended to dominate this area (c++), although advances in compiler technologies has led to languages like rust and go, that manage to show how level of abstraction is distinct from performance.
+
+{: .answer}
+### *&nbsp;*{: .fa .fa-exclamation-circle} Key point: Performance and Level of Abstraction are orthogonal concepts - and writing higher level codes does not make it slower.
+
+Standards:
+
+In vJass, the standard way to write code is the maximally low-level and/or accepted way to implement some function.
+This leads to things like "use array extends when possible" and "prefer unit indexer".
+
+In wurst, the standard way to write code is the maximally *maintainable* way to implement that function.
+This encompasses things like:
+
+* readability
+* documentation
+* comfortable level of abstraction and formatting
+
+*This is actually harder, because the design level thinking is less obvious, and the coder is being pulled in many directions instead of just the standard route.*
+The reason this is better is because *any level of readability improvement, documentation, or comfort* is better than none/flat (as in the case of vJass).
+
+{: .answer}
+### *&nbsp;*{: .fa .fa-exclamation-circle} Key point: Thinking about performance in wurst, and indeed many high level programming languages, is a pitfall, and you have to actively change your behavior to improve your abilities as a programmer.
+
+{: .answer}
+### *&nbsp;*{: .fa .fa-exclamation-circle} Key point: Let the compiler/optimiser do the work for you. Instead, write the code that feels right.
+
+## Is vJass bad then?
+
+For many reasons vJass is still better than plain jass, but JassHelper is inherently a fairly basic preprocessor, so you can't have expectations about it being good enough, and *ought not* to settle for it being the best, or the standard.
+
+More to the point: there was a time when vJass was absolutely the best option for mapping in wc3 - let this be a lesson in economics about being first to market :)
+
+## Is wurst good?
+
+Yes.
+Wurst is more inline with modern, correct programming language design goals.
```

Comment:
"inline" -> "in line". Maybe replace "correct" with "widely accepted" or so, since there are multiple correct solutions in many cases.